File: tools/bench/answers/go_sonnet/s2/13_describe.go
package main

import "fmt"

func describe(v interface{}) string {
	switch val := v.(type) {
	case int:
		return fmt.Sprintf("int %d", val+1)
	case float64:
		return fmt.Sprintf("float %g", val)
	case string:
		return fmt.Sprintf("str %d", len(val))
	}
	return ""
}

func main() {
	fmt.Println(describe(42))
	fmt.Println(describe(3.5))
	fmt.Println(describe("abc"))
}

File: tools/bench/answers/python_haiku/s2/13_describe.py
def describe(v):
    if isinstance(v, bool):
        return f"int {v + 1}"
    elif isinstance(v, int):
        return f"int {v + 1}"
    elif isinstance(v, float):
        return f"float {v}"
    elif isinstance(v, str):
        return f"str {len(v)}"

print(describe(42))
print(describe(3.5))
print(describe("abc"))
